client/client_v0.go

```diff
@@ -25,6 +25,7 @@ func MakeClientV0(settings *types.FireboltSettings, apiEndpoint string) (*Client
 			ClientSecret: settings.ClientSecret,
 			ApiEndpoint:  apiEndpoint,
 			UserAgent:    ConstructUserAgentString(),
+			newVersion:   settings.NewVersion,
```

Comment:
I don't feel too strongly about it, but what do you think about storing settings as a part of the client? This way when NewVersion is discontinued it's easier to remove and also add any additional settings without the need to explicitly propagate them to the base class.

Reply:
I would rather not do this. The reason is I believe we should keep the client as 'thin' as possible. This way we always know what exactly does a client depend on and if this dependency is gone we can safely remove the field. Let me know WDYT